tasks.py

```diff
@@ -497,44 +515,132 @@ def runStartupTasks():
+			"endCursor": query._endCursor,
+			"origKind": query.origKind,
+			"distinct": query._distinct,
+			"classID": cls.__classID__,
+			"customData": customData,
+			"totalCount": 0
+		}
+		cls._requeueStep(qryDict)
+
+	@classmethod
+	def _requeueStep(cls, qryDict: Dict[str, Any]) -> None:
+		"""
+			Internal use only. Pushes a new step defined in qryDict to either the taskqueue or append it to
+			the current request	if we are on the local development server.
+		"""
+		if not queueRegion: # Run tasks inline - hopefully development server
+			req = currentRequest.get()
+			task = lambda *args, **kwargs: cls._qryStep(qryDict)
+			if req:
+				req.pendingTasks.append(task)  # < This property will be only exist on development server!
+				return
+		project = utils.projectID
+		location = queueRegion
+		parent = taskClient.queue_path(project, location, cls.queueName)
+		task = {
+			'app_engine_http_request': {  # Specify the type of request.
+				'http_method': 'POST',
+				'relative_uri': '/_tasks/queryIter'
+			}
+		}
+		task['app_engine_http_request']['body'] = json.dumps(qryDict, cls=JsonKeyEncoder).encode("UTF-8")
+		taskClient.create_task(parent, task)
+
+	@classmethod
+	def _qryStep(cls, qryDict: Dict[str, Any]) -> None:
+		"""
+			Internal use only. Processes one block of five entries from the query defined in qryDict and
+			reschedules the next block.
+		"""
+		from viur.core.skeleton import skeletonByKind
+		qry = db.Query(qryDict["kind"])
+		qry.srcSkel = skeletonByKind(qryDict["srcSkel"])() if qryDict["srcSkel"] else None
+		qry.filters = qryDict["filters"]
+		qry.orders = [(propName, db.SortOrder(sortOrder)) for propName, sortOrder in qryDict["orders"]]
+		qry._startCursor = qryDict["startCursor"]
+		qry._endCursor = qryDict["endCursor"]
+		qry.origKind = qryDict["origKind"]
+		qry._distinct = qryDict["distinct"]
+		if qry.srcSkel:
+			qryIter = qry.fetch(5)
 		else:
-			sharedConf["viur.disabled"] = False
-"""
+			qryIter = qry.run(5)
+		for item in qryIter:
+			try:
+				cls.handleEntry(item, qryDict["customData"])
+			except:
+				sleep(5)
+				try:
+					cls.handleEntry(item, qryDict["customData"])
+				except Exception as e:
+					logging.error("handleEntry failed on %s" % item)
+					logging.exception(e)
+			qryDict["totalCount"] += 1
+		cursor = qry.getCursor()
+		if cursor:
+			qryDict["startCursor"] = cursor
+			cls._requeueStep(qryDict)
+		else:
+			cls.handleFinish(qryDict["totalCount"], qryDict["customData"])
+
+	@classmethod
+	def handleEntry(cls, entry, customData):
+		"""
+			Overridable hook to process one entry. "entry" will be either an db.Entity or an
+			SkeletonInstance (if that query has been created by skel.all())
+
+			Warning: If your query has an sortOrder other than __key__ and you modify that property here
+			it is possible to encounter that object later one *again* (as it may jump behind the current cursor).
+		"""
+		logging.debug("handleEntry called on %s with %s." % (str(cls), entry))
```

Comment:
You could save the `str`-cast. The formatstring will already do this.
```suggestion
		logging.debug("handleEntry called on %s with %s.", cls, entry)
```